src/test/java/org/commonjava/util/partyline/JoinFileWriteAndCloseBeforeFinishedTest.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc. (jdcasey@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.util.partyline;
+
+import org.jboss.byteman.contrib.bmunit.BMScript;
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+
+@RunWith( org.jboss.byteman.contrib.bmunit.BMUnitRunner.class )
+@BMUnitConfig( loadDirectory = "target/test-classes/bmunit", debug = true )
+public class JoinFileWriteAndCloseBeforeFinishedTest
+        extends AbstractJointedIOTest
+{
+
+    @Test
+    @BMScript( "JoinFileWriteAndCloseBeforeFinished.btm" )
+    public void joinFileWriteAndCloseBeforeFinished()
+            throws Exception
+    {
+        final CountDownLatch latch = new CountDownLatch( 2 );
+        final JoinableFile stream = startTimedWrite( 1, latch );
+        startRead( 0, -1, 6000, stream, latch );
```

Comment:
It looks like the rules are managing the signals appropriately, so my previous comment about the Thread.sleep() stuff is a little wrong...but I still think those startTimed*() methods need to be converted to simple start*() methods that spawn a new thread.

Another thing to consider is just using Executors.newCachedThreadPool() and firing off a series of runnable lambda's, like: 

https://github.com/Commonjava/partyline/blob/master/src/test/java/org/commonjava/util/partyline/ConcurrentReadErrorsClearLocksTest.java#L101 

This would cut out all the boilerplate code and also avoid the need for a bunch of complicated inner classes. It wasn't available when I wrote the first tests, because I wasn't using Java 8 yet, but this is how it should have been done all along IMO.